app/views/puppetclasses/_form.html.erb

```diff
@@ -1,5 +1,8 @@
 <%= form_for @puppetclass, :url => (@puppetclass.new_record? ? puppetclasses_path : puppetclass_path(:id => @puppetclass)) do |f| %>
   <%= base_errors_for @puppetclass %>
+  <% unless @puppetclass.errors.empty? %>
```

Comment:
I don't think there can be any error in the class itself, so if there is any error it has to be in the parameters/variables.